src/api/app/controllers/staging/staged_requests_controller.rb

```diff
@@ -1,5 +1,81 @@
-class Staging::StagedRequestsController < Staging::ProjectsController
+class Staging::StagedRequestsController < ApplicationController
+  before_action :require_login, except: [:index]
+  before_action :set_staging_project
+  before_action :set_staging_workflow, :set_project, :set_xml_hash, only: [:create, :destroy]
+
   def index
-    @requests = @project.staged_requests
+    @requests = @staging_project.staged_requests
+  end
+
+  def create
+    authorize @staging_project, :update?
+    requests = @staging_workflow.unassigned_requests.where(number: request_numbers)
+
+    result = ::Staging::StageRequests.new(requests: requests, staging_project_name: @staging_project.name).perform
+    @staging_project.staged_requests << result
+    unassigned_requests = request_numbers - result.pluck(:number).map(&:to_s)
```

Comment:
[nitpick] I would move this to a method as it is repeated several times

Reply:
It's once repeated :thinking: And it is a simple substraction so I don't see the much value in it ...